src/open_inwoner/accounts/views/contactmoments.py

```diff
@@ -70,53 +76,57 @@ def handle_no_permission(self):
             return super().handle_no_permission()
 
 
-class KCMDict(TypedDict):
+@dataclass(frozen=True)
+class ZaakWithApiGroup:
+    zaak: Zaak
+    api_group: ZGWApiGroupConfig
+
+
+class Question(TypedDict):
+    identification: str
+    subject: str
     registered_date: datetime
-    channel: str
-    text: str
-    url: str
-    identificatie: str
-    type: str
-    onderwerp: str
+    question_text: str
+    answer_text: str | None
     status: str
-    antwoord: str
     new_answer_available: bool
+    case_detail_url: str
+    channel: str
 
 
-class KlantContactMomentBaseView(
-    CommonPageMixin, BaseBreadcrumbMixin, KlantContactMomentAccessMixin, TemplateView
-):
-    def get_kcm_data(
-        self,
-        kcm: KlantContactMoment,
-        local_kcm_mapping: dict[str, KlantContactMomentAnswer] | None = None,
-    ) -> KCMDict:
-        data = {
-            "registered_date": kcm.contactmoment.registratiedatum,
-            "channel": kcm.contactmoment.kanaal.title(),
-            "text": kcm.contactmoment.tekst,
-            "url": reverse("cases:contactmoment_detail", kwargs={"kcm_uuid": kcm.uuid}),
-            # eSuite extra
-            "identificatie": kcm.contactmoment.identificatie,
-            "type": kcm.contactmoment.type,
-            "status": Status.safe_label(kcm.contactmoment.status, _("Onbekend")),
-            "antwoord": kcm.contactmoment.antwoord,
-            "new_answer_available": contactmoment_has_new_answer(
-                kcm.contactmoment, local_kcm_mapping=local_kcm_mapping
-            ),
-        }
+class ContextQuestion(Question):
+    identification: str
+    subject: str
+    registered_date: datetime
+    question_text: str
+    answer_text: str | None
+    status: str
+    new_answer_available: bool
 
-        data["onderwerp"] = self.get_kcm_subject(kcm)
 
-        return data
+QuestionValidator = TypeAdapter(Question)
 
-    def get_context_data(self, **kwargs):
-        ctx = super().get_context_data(**kwargs)
-        ctx["anchors"] = self.get_anchors()
-        return ctx
 
-    def get_kcm_subject(
+class MijnVragenService(Protocol):
```

Comment:
"MijnVragenService" might be too narrow, the service could be used elsewhere to fetch `vragen`. Perhaps just `VragenService`?